### PR TITLE
bitflowers: init at 2.0

### PR DESCRIPTION
**Pull request by ethancedwards8**
###### Motivation for this change

Add bitflowers crypto curreny wallet.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ethancedwards8** [comment]
This doesn't build, and its a TOTAL mess. So this is more of a asking for help PR.

This is the repo with the instructions: https://github.com/Crypto-city/bitFlowers

I'm following this: https://nixos.org/manual/nixpkgs/unstable/#sec-language-qt

File: pkgs/applications/blockchains/bitflowers/default.nix [hunk 1]
```diff
@@ -0,0 +1,88 @@
+{ lib, stdenv
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/bitflowers/default.nix:1]:
```suggestion
{ lib
, stdenv
```

File: pkgs/applications/blockchains/bitflowers/default.nix [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    qrencode
+    qtcreator
+    qttools
+    wrapQtAppsHook
+    libtool
+    libglvnd
+    zmqpp
+    protobuf
+    miniupnpc
+    protobufc
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/bitflowers/default.nix:56]: Most of those are probably buildInputs except qmake and wrapQtAppsHook.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -3120,6 +3120,8 @@ in
   birdfont = callPackage ../tools/misc/birdfont { };
   xmlbird = callPackage ../tools/misc/birdfont/xmlbird.nix { stdenv = gccStdenv; };
 
+  bitflowers = libsForQt5.callPackage ../applications/blockchains/bitflowers/default.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:3123]: Are you using qt4 or qt5? You are mixing right now.

Comment (ethancedwards8) [reply pkgs/top-level/all-packages.nix:3123]: The install instructions have both, so to be honest I'm not really sure.

Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:3123]: The install instructions may be out of date. I would suggest try packaging with qt5 only and see how far you can go.

File: pkgs/applications/blockchains/bitflowers/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    leveldb
+    db
+    boost
+  ];
+
+  buildPhase = ''
+    patchShebangs share/genbuild.sh
+    make clean
+    make
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/bitflowers/default.nix:41]:
```suggestion
  preBuild = ''
    patchShebangs share/genbuild.sh
    make clean
  '';
```

**Pull request closed without merging**